bofire/data_models/features/categorical.py

```diff
@@ -363,39 +365,52 @@ class CategoricalOutput(Output):
     order_id: ClassVar[int] = 8
 
     categories: TCategoryVals
-    objective: Annotated[
-        List[Annotated[float, Field(type=float, ge=0, le=1)]], Field(min_items=2)
-    ]
+    objective: Optional[
+        Union[CategoricalObjective, ConstrainedCategoricalObjective]
+    ] = None
 
-    @validator("categories")
+    @validator("categories", allow_reuse=True)
     def validate_categories_unique(cls, categories):
         """validates that categories have unique names
 
         Args:
-            categories (List[str]): List of category names
+            categories (Union[List[str], Tuple[str]]): List or tuple of category names
 
         Raises:
             ValueError: when categories have non-unique names
 
         Returns:
-            List[str]: List of the categories
+            Tuple[str]: Tuple of the categories
         """
-        categories = list(categories)
         if len(categories) != len(set(categories)):
             raise ValueError("categories must be unique")
-        return categories
+        return tuple(categories)
 
     @validator("objective")
-    def validate_objective(cls, objective, values):
-        if len(objective) != len(values["categories"]):
-            raise ValueError("Length of objectives and categories do not match.")
-        for o in objective:
-            if o > 1:
-                raise ValueError("Objective values has to be smaller equal than 1.")
-            if o < 0:
-                raise ValueError("Objective values has to be larger equal than zero")
+    def validate_objectives_unique(cls, objective, values):
+        """validates that categories have unique names
+
+        Args:
+            categories (Union[List[str], Tuple[str]]): List or tuple of category names
+
+        Raises:
+            ValueError: when categories do not match objective categories
+
+        Returns:
+            Tuple[str]: Tuple of the categories
+        """
+        if objective.categories != tuple(values["categories"]):
```

Comment:
We sould have a test for it within `tests/data_models/specs/features.py` using the `add_invalid` syntax.